.github/workflows/docker.yml

```diff
@@ -1,24 +1,68 @@
 # This workflow will upload a Docker container
 
-name: Upload Docker Container
+name: Build/Upload Docker Container
 
 on:
   workflow_dispatch:
   push:
     branches:
-      - 'master'
+      - '*'
+  pull_request:
 
+env:
+  IMAGE: qmk/base_container
 jobs:
   docker:
     runs-on: ubuntu-latest
     steps:
-    - name: Log in to Docker Hub
-      uses: docker/login-action@v1.10.0
-      with:
-        username: ${{ secrets.DOCKER_HUB_USERNAME }}
-        password: ${{ secrets.DOCKER_HUB_ACCESS_TOKEN }}
-    - name: Build and Push to Docker Hub
-      uses: docker/build-push-action@v2.6.1
-      with:
-        push: true
-        tags: qmkfm/base_container:latest
+      - name: checkout
+        uses: actions/checkout@v2
+
+      - name: Cache Docker layers
+        uses: actions/cache@v2
+        with:
+          path: /tmp/.buildx-cache
+          key: ${{ github.ref }}-${{ github.sha }}
+          restore-keys: |
+            ${{ github.ref }}-${{ github.sha }}
+            ${{ github.ref }}
+            refs/head/master
+
+      - name: create tag
+        run: |
+          SHA=${{ github.sha }}
+          TAG=${IMAGE}:$(TZ=UTC-9 date '+%Y%m')-${SHA:0:7}
```

Comment:
Where does `${IMAGE}` come from, and what is its value?

Reply:
Here it is!
https://github.com/ryoppippi/qmk_base_container/blob/437d1781628eb5c72377e1abcdbaa033a19b78e4/.github/workflows/docker.yml#L13

Reply:
Ah, I see. It would be more clear if that were called `CONTAINER_NAME` instead, I was confused about how a lot of it worked because I didn't know what `IMAGE` was.

Reply:
ok I fixed